### PR TITLE
CA-180490: Create a switch to turn on logging for getters.

### PR DESCRIPTION
**Pull request by sharady**

### CODE REVIEW

File: ocaml/xapi/cli_frontend.ml [hunk 1]
```diff
@@ -106,6 +106,15 @@ let rec cmdtable_data : (string*cmd_spec) list =
       flags=[Neverforward];
     };
 
+    "enable-getter-log",
```
Comment (robhoes) [inline ocaml/xapi/cli_frontend.ml]: This command name does not follow the convention of starting with an object name of some sort. I see some existing commands that start with `log-`, so and that seems to fit here.

Also, having `enable` in the command name as well as in the argument is not great. Looking at the existing commands, it is more common to have a pair of `enable`/`disable` commands for this kind of thing.

More generally, isn't it better to do this using a config file? This is how logging is usually controlled.

File: ocaml/idl/ocaml_backend/rbac_audit.ml [hunk 2]
```diff
@@ -396,7 +396,7 @@ let has_to_audit action =
 	let has_side_effect action =
 		not (Xstringext.String.has_substr action ".get") (* TODO: a bit slow? *)
 	in
-	has_side_effect action
+	if not !Xapi_globs.getter_log_enabled then (has_side_effect action) else true
```
Comment (euanh) [inline ocaml/idl/ocaml_backend/rbac_audit.ml]: You can replace this with:
```
!Xapi_globs.getter_log_enabled or (has_side_effect action)
```

File: ocaml/xapi/cli_frontend.ml (hunk 1 above)
Comment (johnelse) [inline ocaml/xapi/cli_frontend.ml]: Seconded that this should go in the config file - otherwise the setting will be lost every time xapi is restarted.

**Comment by sharady** [comment]
Thanks @robhoes @euanh @johnelse  for the suggestions, I have updated the PR to take switch configuration via config file. Please have a look :)

**Comment by simonjbeaumont** [comment]
Removing the label needs-updating. Meaning it has been updated (not that it has been re-reviewed)

**Pull request merged**